src/api/define.js

```diff
@@ -11,38 +11,40 @@ export default function (...args) {
     throw new Error('Skate requires native custom element support or a polyfill.');
   }
 
-  // Support passing an anonymous definition.
-  if (args.length === 1) {
-    // We are checking string for now, but once we remove the ability to pass
-    // an object literal, we can change this to check "function" and invert the
-    // blocks of logic.
-    if (typeof name === 'string') {
-      throw new Error('When passing only one argument to define(), it must be a custom element constructor.');
-    } else {
-      Ctor = name;
-      name = uniqueId();
-    }
+  // DEPRECATED remove when removing the "name" argument.
+  if (DEBUG && args.legnth === 2) {
+    console.warn('The "name" argument to define() is deprecated. Please define a `static is` property on the constructor instead.');
   }
 
-  // Ensure there's no conflicts.
-  if (customElements.get(name)) {
-    name = uniqueId(name);
+  // DEPRECATED remove when removing the "name" argument.
+  if (args.length === 1) {
+    Ctor = name;
+    name = null;
   }
 
-  // DEPRECATED
-  //
-  // Object literals.
+  // DEPRECATED Object literals.
   if (typeof Ctor === 'object') {
     Ctor = Component.extend(Ctor);
   }
 
-  // This allows us to check this before instantiating the custom element to
-  // find its name from the constructor in the vdom module, thus improving
-  // performance but still falling back to a robust method.
-  Ctor[$name] = name;
+  // Ensure a custom element is passed.
+  if (!(Ctor.prototype instanceof HTMLElement)) {
+    throw new Error('You must provide a constructor that extends HTMLElement to define().');
+  }
+
+  // "is" takes precendence over "name" and falls back to a unique id
+  let is = Ctor.is || name || uniqueId();
+
+  // Ensure there's no conflicts.
+  if (customElements.get(is)) {
+    is = uniqueId(is);
+  }
+
+  // Ensure the "is" property is consistent.
+  Ctor.is = is;
```

Comment:
> The point of this isn't for debugging ergonomics, though, it's a nice side-effect. It enables you to do things with the correct element instance that you wouldn't normally be able to do. For example in React without having to use the React integration lib:

yeah I know what is the purpose of `is` , I'm using it everywhere by now :)

> Another idea is to only try and define a unique name if the is property is not provided, thus safe to set. Otherwise it tries to use it and errors if it's already registered. In fact, I like that idea the best, I think. Thoughts?

yup makes sense! that way, we can enforce proper Elements definition by defining `static is`